MahApps.Metro/Controls/WindowSettings.cs

```diff
@@ -64,9 +64,17 @@ public bool UpgradeSettings
         {
             get
             {
-                if (this["UpgradeSettings"] != null)
+                try
                 {
-                    return (bool)this["UpgradeSettings"];
+                    if (this["UpgradeSettings"] != null)
+                    {
+                        return (bool)this["UpgradeSettings"];
+                    }
+                }
+                catch (ConfigurationErrorsException ex)
+                {
+                    var filename = ((ConfigurationErrorsException)ex.InnerException).Filename;
+                    throw new MahAppsException(string.Format("The settings file {0} seems to be corrupted", filename), ex);
                 }
```

Comment:
@Remolutionary ok, really don't know this, so i think this could be a final solution

```cshap
catch (ConfigurationErrorsException ex)
{
    var innerEx = ex.InnerException as ConfigurationErrorsException;
    var filename = innerEx != null ? innerError.Filename : "<unknown>";
    throw new MahAppsException(string.Format("The settings file '{0}' seems to be corrupted", filename), ex.InnerException);
}
```

thx